### DISCUSSION TITLE
⏱ Time & Date update after power loss

### DISCUSSION THREAD

**Discussion by MatthewPoland** [gopro/labs#128] [Q&A]
Hi, is it possible with latest lab firmware to update Time & Date after power loss ?
The H9 is attached to a solar power station (without battery in GoPro), but sometimes the station runs out of juice (ex. 1 week of heavy snow, clouds etc. without direct sun).
After it recharges, GoPro wakes up at USB power and wants to update Time & Date manually or via App.
It starts taking photos, but in wrong time schedule.
My script is `mPNdP>07:59<17:00!SQ~!08:00S!300R`
I have `!MBOOT=Ltimelapse` command to run script and `!MWAKE=2`.
I add `!MSYNC=1`, but it does not update hours 😢
Should I switch to sunrise base command and it should do the trick or wait for next labs firmware with this functionality added ♥️

**Comment by fxstein** [comment]
Don't have the labs answer, although I doubt a camera can update its time without the App or a current QR code.

However, have you considered ignoring the camera date and time and instead using the captured GPS date and time? Its in UTC and pretty much always there. Something like `exiftool` can read the GPS timestamp and write it into the image or video creation date/time of the file. It will require a post-production step but would make the camera independent of any local time source.

**Reply by MatthewPoland** [reply]
How to set up GoPro for adding GPS UTC into exif?
I can't find any other date & time inside image exif information as I paste it below.

**Comment by MatthewPoland** [comment]
Yep, why not.
How to set it up for that?
My last GoPro folder contains all photos with same Date & Time, and I can't say which is taken when.
Exiftool output from random file from today:

ExifTool Version Number | 12.49
-- | --
File Name | GOPR5334.JPG
Directory | /Volumes/NO NAME/DCIM/104GOPRO
File Size | 5.3 MB
File Modification Date/Time | 2022:10:12 14:42:42+02:00
File Access Date/Time | 2022:10:26 00:00:00+02:00
File Inode Change Date/Time | 2022:10:12 14:42:42+02:00
File Permissions | -rwxrwxrwx
File Type | JPEG
File Type Extension | jpg
MIME Type | image/jpeg
Exif Byte Order | Big-endian (Motorola, MM)
Image Description | DCIM\104GOPRO\GOPR5334.JPG
Make | GoPro
Orientation | Horizontal (normal)
X Resolution | 72
Y Resolution | 72
Resolution Unit | inches
Software | HD9.01.01.72.70
Modify Date | 2022:10:12 14:42:41
Y Cb Cr Positioning | Centered
Exposure Time | 1
F Number | 2.5
Exposure Program | Program AE
ISO | 100
Exif Version | 0221
Date/Time Original | 2022:10:12 14:42:41
Create Date | 2022:10:12 14:42:41
Sub Sec Time | 4520
Sub Sec Time Original | 4520
Sub Sec Time Digitized | 4520
Components Configuration | Y, Cb, Cr, -
Compressed Bits Per Pixel | 0.004
Shutter Speed Value | 1
Aperture Value | 2.4
Max Aperture Value | 2.4
Subject Distance | 0 m
Metering Mode | Average
Light Source | Unknown
Flash | No flash function
Focal Length | 3.0 mm
Warning | [minor] Unrecognized MakerNotes
Flashpix Version | ....
Color Space | Unknown (0)
Exif Image Width | 5184
Exif Image Height | 3888
Interoperability Index | R98 - DCF basic file (sRGB)
Interoperability Version | 0100
Exposure Index | 1.066835551
Sensing Method | One-chip color area
File Source | Digital Camera
Scene Type | Directly photographed
Custom Rendered | Normal
Exposure Mode | Auto
Digital Zoom Ratio | 1
Focal Length In 35mm Format | 15 mm
Scene Capture Type | Night
Gain Control | None
Contrast | Normal
Saturation | Normal
Device Setting Description | (Binary data 4 bytes)
Subject Distance Range | Unknown
Serial Number | C34413XXXXXXXX
Compression | JPEG (old-style)
Thumbnail Offset | 44032
Thumbnail Length | 18496
MPF Version | 0100
Number Of Images | 2
MP Image Flags | Dependent child image
MP Image Format | JPEG
MP Image Type | Large Thumbnail (VGA equivalent)
MP Image Length | 292320
MP Image Start | 5044146
Dependent Image 1 Entry Number | 0
Dependent Image 2 Entry Number | 0
Image UID List | (Binary data 66 bytes)
Total Frames | 1
Metadata Version | 8.2.0
Firmware Version | HD9.01.01.72.70
Camera Serial Number | C34413XXXXXXXX
Camera Model Name | HERO9 Black
Media Unique ID | 67aed47653efe4b77d1caddd38d1693100000000000000000000000000000000
Auto Rotation | Up
Digital Zoom | Yes
Pro Tune | On
White Balance | 4500K
Sharpness | LOW
Color Mode | GOPRO
Maximum Shutter Angle | AUTO
Auto ISO Max | 100
Auto ISO Min | 100
Exposure Compensation | 0.0
Rate | 1
Photo Resolution | 20MP_W
HDR Setting | OFF
Device Name | User Metadata
Image Width | 5184
Image Height | 3888
Encoding Process | Baseline DCT, Huffman coding
Bits Per Sample | 8
Color Components | 3
Y Cb Cr Sub Sampling | YCbCr4:2:2 (2 1)
Aperture | 2.5
Shutter Speed | 1
Create Date | 2022:10:12 14:42:41.4520
Date/Time Original | 2022:10:12 14:42:41.4520
Modify Date | 2022:10:12 14:42:41.4520
Thumbnail Image | (Binary data 18496 bytes)
Preview Image | (Binary data 292320 bytes)
Image Size | 5184x3888
Light Value | 2.6
Megapixels | 20.2
Scale Factor To 35 mm Equivalent | 5.0
Circle Of Confusion | 0.006 mm
Field Of View | 100.4 deg
Focal Length | 3.0 mm (35 mm equivalent: 15.0 mm)
Hyperfocal Distance | 0.60 m

and it's for sure not a photo from 12.10.2022 14:42

**Comment by catch-point** [comment]
This might help: https://gopro.github.io/labs/control/gpssync/

**Reply by MatthewPoland** [reply]
Hi, as I wrote earlier 
> I add `!MSYNC=1`, but it does not update hours

If GoPro can still wake up at specific time and the only downside of not having date & time updated is wrong image creation date & time – it's fine with me.
I can do the post work to dig out the exif information from the image.
But now I don't know if it's running properly and the image don't have this exif information as well.

**Reply by fxstein** [reply]
@jamesrdf It would be amazing if labs could add an extension to the sync feature, like a second sync mode that sets the entire date and time, not just minutes and seconds.

Something like `!MSYNC=2` that sets the entire time information from GPS - yes in UTC in this case.

**Comment by fxstein** [comment]
Here you go. For photos its easier than for videos, but possible for both.

Run this to see all the time info of the files:

```
exiftool -time:all -GPSDateTime -s P_20221004122643_GoPro_Hero10_2442_G0280333.jpg 
```

I am in the middle of writing a little feature that does exactly that when importing images from the sdcard. It will be something like

```
goprox --import:gps
```

but have not gotten around to it just yet. Working on the GoPro video time shift to UTC at the moment - as all GoPro video files come with the wrong time by default. If interested follow along here: https://github.com/fxstein/GoProX

Hope this helps!

**Reply by fxstein** [reply]
Forgot to mention that all GPS times will be UTC. But even that is no problem. I wrote a geonames feature that will take the first file of a day and perform a GPS based timezone lookup that can then be applied to all files of the day.

**Comment by MatthewPoland** [comment]
Thanks for your tip, I will try to restore date & time via exiftool when I will have access to images.
But what about starting and ending the code?
I assume that GoPro – without date & time updated – will randomly start the qr code script and end accordingly to the time shift that is in camera after power loss, right?
